### DISCUSSION TITLE
Issue with serializing array of doubles.

### DISCUSSION THREAD

**Discussion by NateTonn** [nlohmann/json#4202] [Q&A]
Hi there all, 

Some introductory information: I am new here and to this json library, so this might be a simple fix, but I couldn't find information about it. I am using the library with IAR EW 9.30.1. 
I am having issues with serializing an array of floating-point numbers, both float and double. This library has been working well for me and my purposes so far. I just hit this issue today and was wondering if anyone might know what is going wrong.

So, I shrunk the example down for the sake of compactness, but this happens with any size array. 
`
json jsonObject;

std::vector<double> doubleArray = {10.0, 15.0, 20.0};

std::vector<uint32_t> intArray = {0, 1, 2};  

jsonObject["doubles"] = doubleArray;

jsonObject["ints"] = intArray;

std::string testString = jsonObject.dump();
`
I can set breakpoints and see that the vectors, and jsonObject have the proper values in them. But after using dump(), the resulting string looks like this:
{"doubles":[%g.0,%g.0,%g.0], "ints":[0,1,2]}
So, the integers work perfectly, but I get these strange values for when the doubles are serialized. Every single double/float no matter the value gets set to %g.0 and I have no idea why. If anyone has a clue, I would appreciate it.

**Comment by gregmarr** [comment]
If that's your exact code, then your compiler must be doing something wrong.  That works fine on Clang and GCC.  https://compiler-explorer.com/z/vzahxrnc7

**Reply by NateTonn** [reply]
That is my exact code, so I assume that it is the compiler. The IDE has its own compiler, the IAR C/C++ compiler, and it isn't on that website. I can't imagine why this is causing issues, the compiler has never had issues with libraries like this before. Do you think the library is just incompatible with the compiler? Or could there be some sort of configuration setting in the library that I could change? I kind of have to use IAR in this case, so switching isn't really an option.

**Reply by gregmarr** [reply]
The only thing I can think of is that the library somehow doesn't support the `%g` format specifier, which would be really odd.

**Reply by gregmarr** [reply]
Looks like that's the case: https://forums.mbed.com/t/how-to-use-standard-version-of-printf-snprintf-on-iar-ewarm/16105
